### PR TITLE
add error type to distinguish btw metadata and file error

### PR DESCRIPTION
**Pull request by prabh-t**

### CODE REVIEW

**Comment by yusra-haider** [comment]
@prabh-t could you please add examples of the different error types that we could have for different cases in the description? Truncated examples of the documents that would get created in MongoDB would be super helpful!

**Comment by yusra-haider** [comment]
Would we be needing a migration script for this?

**Comment by prabh-t** [comment]
> @prabh-t could you please add examples of the different error types that we could have for different cases in the description? Truncated examples of the documents that would get created in MongoDB would be super helpful!

@yusra-haider The new property in the validation error object will be `errorType` with one of the following error type strings:
```
METADATA_ERROR, 
FILE_NOT_UPLOADED,
FILE_ERROR
```
Yes, we need to migrate existing file documents with validation errors.

File: src/listener/handlers/document-update-handler.ts [hunk 1]
```diff
@@ -129,9 +130,9 @@ class DocumentUpdateHandler implements IHandler {
                         return this.attemptFileValidation(contentValidationReport, doc, documentType) 
                     } else {
                         // otherwise set validation state to INVALID, return error report with NoCloudUrl
-                        const msg = "Valid metadata. File is not uploaded.";
-                        const err = new ErrorReport(msg);
-                        err.userFriendlyMessage = msg;
+                        const msg = "File cloudUrl property not set.";
+                        const err = new ErrorReport(ErrorType.FileNotUploaded, msg);
+                        err.userFriendlyMessage = "File not uploaded.";
```
Comment (aaclan-ebi) [inline src/listener/handlers/document-update-handler.ts:135]: "The file is not uploaded yet." @gabsie or @clairerye might have a better suggestion.

**Review by aaclan-ebi — APPROVED** [review]
Looks good!

**Pull request merged**